### ISSUE TITLE
MQTTNet server - Instructions on build or are there pre-built binaries?

### ISSUE THREAD

**Issue by WattsC-90** [chkr1011/MQTTnet#859]
### Describe your question
I would like to use the MQTTNet Server for testing and general R&D while working with the MQTTClient (Managed). I dont appear to be able to get any binaries from the releases page (for windows), and I cant build the source code for 3.0.8 in 2019VS.. do you have any instructions for build?
I get an issue with nuget packages, that the package `Microsoft.AspNetCore.Connections.Abstractions` has two conflicting versions and 1 cannot be updated as it locked by the project. I have not seen this before and even after looking on google am none the wiser as to how to solve it.

Does anyone know how to get the project building for MQTTNet Server please? Regards

**Comment by SeppPenner** [comment]
What do you mean with binaries? The nuget package(s) as dll file(s)? Or do you want a simple running MQTT server for testing as binaries?

If the first applies, I can give you the NuGet packages and you can install them locally.

For the second one, I have an example project under https://github.com/SeppPenner/NetCoreMQTTExampleJsonConfig. I can provide you binaries for this and it can be configured quite well.

**Comment by WattsC-90** [comment]
@SeppPenner no certainly not the nugets, I have been looking into those from nuget.org already, but the MQTTnet Server is something i was very keen on looking at/using. I couldnt find a built binary and after downloading the source, I am unable to build the MQTTNet Server project at all? My understanding was that the MQTTNet Server project was an MQTT broker, so was quite keen to investigate.
Thanks for your help

**Comment by SeppPenner** [comment]
@WattsC-90 Ok, I agree with you. When I build the project from 3.0.8, I get the following error:

```log
Fehler	MSB4019	Das importierte Projekt "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\WindowsXaml\v16.0\Microsoft.Windows.UI.Xaml.CSharp.targets" wurde nicht gefunden.
Stellen Sie sicher, dass der Ausdruck in der Import-Deklaration "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\WindowsXaml\v16.0\Microsoft.Windows.UI.Xaml.CSharp.targets"
richtig ist und dass die Datei auf dem Datenträger vorhanden ist.	MQTTnet	C:\Program Files\dotnet\sdk\3.1.101\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.targets
```

I know, it's German (sorry :D), but this seems to be the same error as described under https://stackoverflow.com/questions/45041972/microsoft-windows-ui-xaml-csharp-targets-missing-on-ci-server.

@JanEggers @chkr1011 Do you have any fixes here?

So, I cannot build the broker either. I can give you my project which utilizes the broker as I told you.

**Comment by SeppPenner** [comment]
I have built a simple server application that can be configured with port and username, clientid and password for multiple users: https://github.com/SeppPenner/SimpleMqttServer. There you can find a release (That needs NetCore 3.1.101 installed). You can build the project (and modify it if you like), of course.

**Comment by WattsC-90** [comment]
I just wanted a binary to run to be honest, im more interested in trying out the managed client and it seemed good fortune that it had everything bundled together! Ill have a look at  your project, thanks for your help!

**Comment by SeppPenner** [comment]
Well, there is no binary for the broker itself as far as I know. You can checkout the project and use the binaries to test the managed client with it or build it on your own :)

**Comment by WattsC-90** [comment]
@SeppPenner , I would build it on my own if it compiled haha

**Comment by SeppPenner** [comment]
Yeah sure. I meant, you could compile my project from source :)

**Comment by biapar** [comment]
https://github.com/chkr1011/MQTTnet/releases/download/v3.0.7/MQTTnet.Server-Windows-x64-v3.0.7.zip

**Comment by SeppPenner** [comment]
@biapar You're right. Normally, the latest version should be in the releases section. However, there is non (yet) for 3.0.8.

@chkr1011 This is something for you.

**Comment by SeppPenner** [comment]
So, there was an issue with the releases (Check https://github.com/chkr1011/MQTTnet/issues/850#issuecomment-600097186).

As @biapar pointed out, this normally can be found under the releases (And will be available again for 3.0.9 soon).
